pip/utils/__init__.py

```diff
@@ -105,28 +105,31 @@ def find_command(cmd, paths=None, pathext=None):
     # check if there are funny path extensions for executables, e.g. Windows
     if pathext is None:
         pathext = get_pathext()
-    pathext = [ext for ext in pathext.lower().split(os.pathsep) if len(ext)]
+    pathext = [ext for ext in pathext.lower().split(os.pathsep) if ext]
     # don't use extensions if the command ends with one of them
     if os.path.splitext(cmd)[1].lower() in pathext:
-        pathext = ['']
+        pathext = []
```

Comment:
Why does this not mean that on Unix, commands with no extension won't be found? It's not at all obvious from the code.